src/sorted-table.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import {useState} from "react";
+import "./sorted-table.css";
+
+export default function SortedTable({tableData}) {
+
+    const tableContentWithKey = () => tableData.slice(1).map((row, i) => [i].concat(row));
+    const compare = (a,b) =>
+        typeof a === "number" ? a - b
+            : typeof a === "string" ? (a.toLowerCase() < b.toLowerCase() ? -1 : a.toLowerCase() > b.toLowerCase() ? 1 : 0)
```

Comment:
Why do you test for string/boolean?
It is hard to read because all your sorting functions use `a` and `b` as variable names.

Reply:
How would you name two variables of a compare function?

Reply:
Call them by what it is you are comparing, `tableRow1` for example

Reply:
I do not agree in this case. It is a generic comparison function that compares two parameters.
Maybe it should reside elsewhere. I moved it outside of the component so maybe it is clearer now;

Reply:
This function is not generic at all. You use it for something very specific in this project. You can't just copy paste it into another without modifications.
There will be very rare cases that you will actually use a compare function so generic that it can take any possible value. In this case you try to make it even more generic by checking for numbers which simply do not exist in the data.

Moreover, the idea of using a function so generic to solve a specific issue with specific data, is the wrong approach. Never try to solve problems that don't yet exist.